dagster/pandas_kernel/__init__.py

```diff
@@ -97,3 +100,36 @@ def single_path_arg(input_name, path):
 
 def json_input(name):
     return create_json_input(name)
+
+
+# XXX(freiksenet): this should really go to main solid file
+def validate_transform_fn(solid_name, transform_fn, inputs):
+    input_names = set(inp.name for inp in inputs)
+    used_inputs = set()
+    has_kwargs = False
+
+    signature = inspect.signature(transform_fn)
+    for i, param in enumerate(signature.parameters.values()):
+        if param.kind == inspect.Parameter.VAR_KEYWORD:
+            has_kwargs = True
+        elif param.kind == inspect.Parameter.VAR_POSITIONAL:
+            check.failed(
+                f"solid '{solid_name}' transform function has positional vararg parameter '{param.name}'. Transform functions should only have keyword arguments that match input names and optionally a first positional parameter named 'context'."
+            )
+        else:
+            # XXX(freiksenet): I don't like this
```

Comment:
I would prefer that you just do something like create a parameter of dictionaries by param name filter out the context variable there. All variables are passed by kwarg so this will break valid transform functions.

Reply:
done